### ISSUE TITLE
[Object Detection][Bug] keep_aspect_ratio_resizer doesn't work well with pad_to_max_dimension

### ISSUE THREAD

**Issue by netanel-s** [tensorflow/models#5388]
### System information
- **What is the top-level directory of the model you are using**: object_detection
- **Have I written custom code**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.10.
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**: CUDA 9
- **GPU model and memory**: 1080Ti, 12GB
- **Exact command to reproduce**: training and evaluating ssdlite_mobilenet_v2_coco_2018_05_09 using object_detection/model_main.py.

### Describe the problem
For a while now I tried to train a model using keep_aspect_ratio_resizer without any luck. I believe I found that the reason is that this resizer simply doesn't work well.
Here's a keep_aspect_ratio_resizer configuration I've been used:
```
      keep_aspect_ratio_resizer {
        min_dimension: 480
        max_dimension: 640
        pad_to_max_dimension: true
      }
```
The way I expected it to work is to resize the image such that the longer axis will have length 640 while keeping the original aspect ratio, and then the image is padded (right-bottom) in order to achieve 640x640. Moreover, the bounding boxes should be resized in the same manner.
BTW, the reason I use the padding is to have batch_size>1. Otherwise, multiple images which don't have the same dimensions won't fit together to a single tensor.

Here you can see visualization of GT with **keep_aspect_ratio_resizer** and **pad_to_max_dimension** (note how the aspect ratio is not kept, and that the BBs have the original shape and location):
![image](https://user-images.githubusercontent.com/38940293/46131974-9ef21f80-c245-11e8-84d8-2d010278e40c.png)
![image](https://user-images.githubusercontent.com/38940293/46132022-bb8e5780-c245-11e8-9dbc-816708f1c7b4.png)
and here the same images with fixed_shape_resizer:
![image](https://user-images.githubusercontent.com/38940293/46132047-cba63700-c245-11e8-924e-3744b588e832.png)
![image](https://user-images.githubusercontent.com/38940293/46132055-d1038180-c245-11e8-8ef1-ac14866fa3f4.png)

Thanks in advance,

**Comment by netanel-s** [comment]
@derekjchow , do you have a useful insight which can help?
I don't mind changing the code myself, but I'm not sure what's the bug.
I tried changing the [returned size](https://github.com/tensorflow/models/blob/master/research/object_detection/core/preprocessor.py#L2218) in case pad_to_max_dimension is true to [max_dim, max_dim, 3], but I'm not sure it fixed the bug.

**Comment by SimsGautam** [comment]
@netanel-s @derekjchow Any resolution to this bug or new directions?

**Comment by edumotya** [comment]
I have the same problem. I think it is obvious, the function [resize_to_range](https://github.com/tensorflow/models/blob/master/research/object_detection/core/preprocessor.py#L2212) pad the image but let the bounding boxes unchanged, actually this function does not have access to the bounding boxes. However, we should add an offset to the normalized bounding box coordinates when padding the image.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
